src/commands/config/create.ts

```diff
@@ -0,0 +1,88 @@
+import fs from "fs";
+import { error, info, success } from "signale";
+import { Argv } from "yargs";
+import { deployDocumentStore } from "../../implementations/deploy/document-store/document-store";
+import { deployTokenRegistry } from "../../implementations/deploy/token-registry/token-registry";
+import { readFile } from "../../implementations/utils/disk";
+import { create as CreateWallet } from "../../implementations/wallet/create";
+import { getLogger } from "../../logger";
+import { handler as CreateTemporaryDns } from "../dns/txt-record/create";
+import ConfigTemplate from "./config-template.json";
+import { CreateConfigCommand } from "./config.type";
+
+const { trace } = getLogger("config:create");
+
+export const command = "create [options]";
+
+export const describe = "Create a generic config file";
+
+export const builder = (yargs: Argv): Argv =>
+  yargs
+    .option("output-dir", {
+      alias: "od",
+      description: "Write output to a directory",
+      type: "string",
+      demandOption: true,
+    })
+    .option("encrypted-wallet-path", {
+      alias: "ewp",
+      type: "string",
+      description: "Path to file containing wallet.json",
+      normalize: true,
+      demandOption: false,
+    });
+
+export const handler = async (args: CreateConfigCommand): Promise<void> => {
+  trace(`Args: ${JSON.stringify(args, null, 2)}`);
+  if (!fs.existsSync(args.outputDir)) {
+    fs.mkdirSync(args.outputDir);
+  }
+
+  try {
+    let walletPath = "";
+    if (!args.encryptedWalletPath) {
+      info(`Enter password to create wallet`);
+      args.outputFile = `${args.outputDir}/wallet.json`;
```

Comment:
99,99999999999999999% of the time it's a bad idea to alter an object

```
const x = { a:"foo"}
const fn = (o) => { o.b = "g" }
fn(x)
// x has now 2 properties, the function is impure (https://stackoverflow.com/questions/35825076/what-is-an-example-of-an-impure-function-in-javascript)
```

dont alter args. create temporary variables and provides them to the function that needs it

Reply:
ok will do